### ISSUE TITLE
Impossible to specify a custom worker instance type in the autoscaling dialog box

### ISSUE THREAD

**Issue by hackdna** [galaxyproject/cloudman#39]
AMI: ami-d5246abf, Name: Galaxy-CloudMan-1449500413

There is no input box to type in an actual custom instance type (as it is possible when adding worker nodes manually):
![screenshot 2016-02-12 12 38 16](https://cloud.githubusercontent.com/assets/452575/13015299/4e103cda-d186-11e5-8d55-c58ba7e57508.png)

**Comment by hackdna** [comment]
Also, worker nodes are not added with autoscaling on:
```
galaxy.jobs.runners.drmaa DEBUG 2016-02-12 18:20:44,331 (350) native specification is: -n 8 --ntasks-per-node=8
galaxy.jobs.runners.drmaa WARNING 2016-02-12 18:20:44,338 (350) drmaa.Session.runJob() failed, will retry: code 1: slurm_submit_batch_job: More processors requested than permitted
```

**Comment by afgane** [comment]
Thanks for reporting the lack of a custom instance type input field. It's fixed in 87949d97c639aebf01e4fc2e732b445d9f886e07. Until we make the next CloudMan release, you can use the fixed version for any new cluster by specifying `cloudman-test` as the value for the default bucket under the advanced options on launch.usegalaxy.org.

Re. the worker nodes not being added - under what circumstances is that? Did you mofidy Galaxy's `job_conf.xml`?

**Comment by hackdna** [comment]
Great, thanks!

Yes, I've added the following to `job_conf.xml` to allow bowtie2 use more than the default one core:
```
<destination id="bowtie2" runner="slurm>
    <param id="nativeSpecification">-n 8 --ntasks-per-node=8</param>
</destination>
<tool id="toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2" destination="bowtie2"/>
```

**Comment by afgane** [comment]
Are you sure one of those instance types was available? I just tried it with that exact Galaxy setup and only one c3.2xlarge instance in a cluster and a job ran fine.

**Comment by hackdna** [comment]
Here's my setup: min worker nodes - zero, max worker nodes - one, worker node type: c3.2xlarge, running jobs on master (c3.large) is turned off. No modifications to `job_conf.xml`.

When trying to run the fastq_groomer tool no worker nodes get started, the corresponding history item says "This job is waiting to run" and `main.log` shows the following:
```
galaxy.jobs.runners.drmaa INFO 2016-02-16 22:19:56,137 (346) queued as 8
galaxy.jobs DEBUG 2016-02-16 22:19:56,164 (346) Persisting job destination (destination id: slurm_cluster)
galaxy.jobs.runners.drmaa DEBUG 2016-02-16 22:19:56,815 (346/8) state change: job is queued and active
```

**Comment by hackdna** [comment]
Looks like exactly the same problem: http://dev.list.galaxyproject.org/CloudMan-Autoscaling-Unable-to-run-this-job-due-to-a-cluster-error-tt4665748.html

**Comment by afgane** [comment]
I have something to attend to now but I'll fix this by the end of the day.

**Comment by afgane** [comment]
OK - should be fixed now. Let me know if you find anything else that's off.

**Comment by hackdna** [comment]
Thank you for implementing a fix so quickly. I've tried testing it by specifying `cloudman-test` and `cloudman-dev` as the "Default bucket" at https://launch.usegalaxy.org/launch ([as described](https://wiki.galaxyproject.org/CloudMan/Building#Get_your_own_bucket)) but for some reason the master node still has the old code. Any idea what could be wrong?

**Comment by afgane** [comment]
Did you use a cluster name for a cluster that you created before and had shut down? In that case (for reproducibility reasons), the original code will be reused. If that was the case, try with a new cluster name.

**Comment by hackdna** [comment]
Thank you, I was reusing the cluster name. Creating a new cluster with a new name using cloudman-test bucket worked as expected. I did see this error message in the cloudman log during startup:
```
22:07:39 - The master instance is set to *not* execute jobs. To manually change this, use the CloudMan Admin panel.
22:07:39 - Adding on-demand instance(s)
22:07:39 - EC2 response error when starting worker nodes: EC2ResponseError: 400 Bad Request InvalidParameterValueMaximum instance count must be greater than zero492d932a-1f2f-47c1-9aee-fdf17f55c1bb
```

**Comment by afgane** [comment]
I believe this is also fixed with 085ba1657fcfcd1e522f1505fa96aca7adf8fabe.

**Comment by hackdna** [comment]
Thanks!
